andela/knights-bn-backend
# #170947679 Send notification to the manager when request is edited

#### What does this PR do?
- Send notification to the user and his/her manager when a request is modified
#### Description of Task to be completed?
- When a requester modifies something on the trip request he/she has made, notification will be
pop up and notification panel will be updated to both requester and his/ her manager, the notification will be clickable and redirect the user to the edited trip request.

#### How should this be manually tested?
- open your terminal and run this command `git clone https://github.com/andela/knightsbn    backend.git` to clone repository
- run `cd knights-bn-backend` to get inside the project
- run `git fetch && git checkout ft-receive-notification-when-edit-170947679` 
- run `npm install` to install dependencies
- check if you have PostgreSQL database if you don't have it install it from here [PostgreSQL](https://www.2ndquadrant.com/en/blog/pginstaller-install-postgresql/)
- create database
- create `.env` file in your project root directory
- follow `env.example` file to create environment variables in `.env` file and 
   assign them with all your database configurations 
- run `npm run createTables` to create tables in the new database you have created.
- run `npm run dev-start` to start the server.
- open your postman or any other API tester you have and use this URL to login `http://localhost:4000/api/v1/auth/login` and down below there is a default user you can use
```
{
   "email": "alain.maxime@gmail.com",
    "password": "Niyonkuru@1"
}
```
- open this URL in your browser `http://localhost:4000/client/editRequest.html`
allow notification permission first by clicking on allow button on the browser alert which looks like the image below, you can also use this page to view all the trip request you have so that you can edit them.
<img width="306" alt="Screen Shot 2020-03-17 at 07 56 30" src="https://user-images.githubusercontent.com/36619897/76826377-04b0dd80-6825-11ea-938c-5d2ea3662463.png">

- use postman and send PATCH request to this URL`http://localhost:4000/api/v1/trips/edit/{id}` to update trip request of your choice.

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170947679](https://www.pivotaltracker.com/story/show/170947679)
#### Screenshots
#### Notification
<img width="332" alt="Screen Shot 2020-03-17 at 13 03 50" src="https://user-images.githubusercontent.com/36619897/76851082-55d6c680-6851-11ea-8b0a-21d84777ce2e.png">


#### Update notification panel
<img width="1113" alt="Screen Shot 2020-03-17 at 13 03 36" src="https://user-images.githubusercontent.com/36619897/76851127-64bd7900-6851-11ea-8364-07dbb0823298.png">


#### Redirect to the trip request
<img width="1293" alt="Screen Shot 2020-03-13 at 09 26 12" src="https://user-images.githubusercontent.com/36619897/76851202-8d457300-6851-11ea-8b0b-1d3868f5c79f.png">
